python/pyspark/java_gateway.py

```diff
@@ -51,13 +51,16 @@ def launch_gateway():
         on_windows = platform.system() == "Windows"
         script = "./bin/spark-submit.cmd" if on_windows else "./bin/spark-submit"
         submit_args = os.environ.get("PYSPARK_SUBMIT_ARGS", "pyspark-shell")
+        if conf and conf.getAll():
+            submit_args = ' '.join(['--conf %s="%s"' % (k, v) for k, v in conf.getAll()]) \
```

Comment:
I think it would be better to create a separate list for the `--conf` arguments, and concatenate it into the `command` variable below. Then you don't need to worry about the values of the arguments causing trouble.